sdk/cosmosdb/cosmos/src/queryExecutionContext/ContinuationTokenManager/BaseContinuationTokenManager.ts

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { QueryRangeMapping } from "../queryRangeMapping.js";
+import type {
+  QueryRangeWithContinuationToken,
+  SimplifiedQueryRange,
+} from "../../documents/ContinuationToken/CompositeQueryContinuationToken.js";
+import type {
+  PartitionRangeUpdate,
+  PartitionRangeUpdates,
+} from "../../documents/ContinuationToken/PartitionRangeUpdate.js";
+import { PartitionRangeManager } from "../PartitionRangeManager.js";
+
+/**
+ * Interface representing the result portion of a query response that contains
+ * continuation token related data.
+ * @internal
+ */
+export interface QueryResponseResult {
+  partitionKeyRangeMap?: Map<string, QueryRangeMapping>;
+  updatedContinuationRanges?: PartitionRangeUpdates;
+  orderByItems?: { orderByItems: any[]; _rid: string }[];
+  buffer?: any[];
+}
+
+/**
+ * Base abstract class for continuation token management.
+ * Provides common functionality shared between parallel and ORDER BY query token managers.
+ * @internal
+ */
+export abstract class BaseContinuationTokenManager {
+  protected ranges: QueryRangeWithContinuationToken[] = [];
+  protected partitionRangeManager: PartitionRangeManager = new PartitionRangeManager();
+  protected isUnsupportedQueryType: boolean = false;
```

Comment:
can we remove this param and avoid creating token manager for unsupported queries.

Reply:
addressed